api/v1alpha1/scheduledscaling_types.go

```diff
@@ -0,0 +1,132 @@
+
+// ScheduledScalingSpec defines the desired state of ScheduledScaling
+type ScheduledScalingSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	/// The schedule in Cron format, see https://en.wikipedia.org/wiki/Cron.
+	Schedule Schedule `json:"schedule" protobuf:"bytes,1,name=schedule"`
+
+	// TargetRef is the targets that need to be scheduled
+	TargetRefs TargetRefs `json:"targetRefs" protobuf:"bytes,2,name=targetRefs"`
+
+	// Currently only static, might implement others(?)
+	Strategy Strategy `json:"strategy" protobuf:"bytes,3,name=strategy"`
+}
+
+// ScheduledScalingStatus defines the observed state of ScheduledScaling
+type ScheduledScalingStatus struct {
+	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	ScheduledScalingPhase ScheduledScalingPhase `json:"scheduledScalingPhase" protobuf:"bytes,1,name=scheduledScalingPhase"`
+}
+
+type ScheduledScalingPhase string
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+// ScheduledScaling is the Schema for the scheduledscalings API
+type ScheduledScaling struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   ScheduledScalingSpec   `json:"spec,omitempty"`
+	Status ScheduledScalingStatus `json:"status,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+
+// ScheduledScalingList contains a list of ScheduledScaling
+type ScheduledScalingList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []ScheduledScaling `json:"items"`
+}
+
+type TargetRefs struct {
+	// ScaleTargetRef is the target of scaling.
+	// It should be the same as the target of HPA.
+	ScaleTargetRef CrossVersionObjectReference `json:"scaleTargetRef" protobuf:"bytes,1,name=scaleTargetRef"`
+	//Tortoise to be targeted for scheduled scaling
+	TortoiseName *string `json:"tortoiseName,omitempty" protobuf:"bytes,2,name=tortoiseName"`
+}
+
+// CrossVersionObjectReference contains enough information toet identify the referred resource.
+type CrossVersionObjectReference struct {
+	// kind is the kind of the referent; More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds
+	Kind *string `json:"kind,omitempty" protobuf:"bytes,1,opt,name=kind"`
+
+	// name is the name of the referent; More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names
+	Name *string `json:"name,omitempty" protobuf:"bytes,2,opt,name=name"`
+
+	// apiVersion is the API version of the referent
+	// +optional
+	APIVersion *string `json:"apiVersion,omitempty" protobuf:"bytes,3,opt,name=apiVersion"`
+}
+
+type Schedule struct {
+	/// The schedule in Cron format, see https://en.wikipedia.org/wiki/Cron.
```

Comment:
It's not a cron format. It should represent date plus time.

Reply:
I think we can just use RFC3339 format. 
https://datatracker.ietf.org/doc/html/rfc3339

e.g., `"2006-01-02T15:04:05Z09:00"`

Not sure whether it's user-friendly or not though... But I cannot think of any other easier representation.